src/ButterCMS/ButterCMS.php

```diff
@@ -24,18 +27,23 @@ public function __construct($authToken)
     {
         $this->authToken = $authToken;
 
-        $this->client = new Client();
+        $this->client = new Client([
+            'headers' => [
+                'X-Butter-Client' => 'PHP/2.3.0',
+            ],
+        ]);
     }
 
     protected function request($url, $params = [], $tryCount = 0)
     {
         try {
             $params['auth_token'] = $this->authToken;
             $response = $this->client->get(self::API_ROOT_URL . $url, [
-                'query' => $params
+                'query' => $params,
             ]);
-        } catch (ClientException $e) {
-            if ($tryCount < 1) {
+        } catch (BadResponseException $e) {
+            $httpCode = (int)$e->getResponse()->getStatusCode();
+            if ($tryCount < 1 && $httpCode !== 404) {
```

Comment:
Can you explain this logic?

Reply:
The `$tryCount < 1` was there previously for a simple retry. We added that previously for some reason, I forget why.

I added the 404 check because I assume if the URL is a 404 that's not going to change with an immediate retry so we might as well avoid a second round trip call to the API. Basically, if I make a call to `fetchPages('non-existent-category')`, there's no point retrying that.

Reply:
Got it, makes sense thanks